src/app/api/updateTask/[id]/route.ts

```diff
@@ -0,0 +1,39 @@
+import prisma from "@/lib/prisma";
+import { z } from "zod";
+
+const ZTaskSchema = z.object({
+  id: z.number().optional(),
+  title: z.string(),
+  description: z.string(),
+  category: z.string(),
+  date: z.string().datetime(),
+  userId: z.string(),
+});
+export async function PATCH(
+  request: Request,
+  { params }: { params: { id: string } }
+) {
+  try {
+    const formData = await request.formData();
+    const task = ZTaskSchema.parse({
+      title: formData.get("title") as string,
+      description: formData.get("description") as string,
+      category: formData.get("category") as string,
+      date: formData.get("date") as string,
+      userId: formData.get("userId") as string,
+    });
+    const updatedTask = await prisma.task.update({
+      where: {
+        id: parseInt(params.id, 10),
+      },
+      data: {
```

Comment:
```suggestion
      data: task
```

Do you need the spread here?

Reply:
I remove spread and its not necessary